core/src/main/java/org/jivesoftware/sparkimpl/certificates/SparkTrustManager.java

```diff
@@ -187,6 +188,17 @@ public void checkServerTrusted(X509Certificate[] chain, String authType) throws
     }
 
     /**
+     * Return true if the certificate chain contain only one Self Signed certificate
+     */
+    private boolean isSelfSigned(X509Certificate[] chain) {
```

Comment:
I added check if chain length equals 1. If it is true together as well as same subject issuer requirement then certificate is self signed. This way in case when chain would be reversed there will be no problems about that. 
One note: I don't worry here about standard root CA at the end of certificate path but about situation when server send only one self signed certificate for it's identification. If chain is longer than 1 then it is standard path building.